dsgrid/registry/project_registry_manager.py

```diff
@@ -239,36 +247,84 @@ def _submit_dataset(
-    #    project_config : ProjectConfig
-    #    dataset_config : DatasetConfig
-    #    dataset : Dataset
-
-    #    Raises
-    #    ------
-    #    DSGInvalidDimensionMapping
-    #        Raised if a requirement is violated.
-
-    #    """
-    #    needs_full_join = set()
-    #    types = [x for x in DimensionType if x != DimensionType.TIME]
-    #    for type1, type2 in itertools.product(types, types):
-    #        if type1 != type2:
-    #            needs_full_join.add((type1, type2))
-
-    #    for mapping_ref in project_config.model.dimension_mappings.base_to_base:
-    #        from_dimension = mapping_ref.from_dimension_type
-    #        to_dimension = mapping_ref.to_dimension_type
-    #        needs_full_join.remove((from_dimension, to_dimension))
-    #        mapping_config = self._dimension_mapping_mgr.get_by_id(mapping_ref.mapping_id)
-    #        mapping_records = models_to_dataframe(mapping_config.model.records)
-    #        # TODO perform checks across mapping_records, dataset.load_data, dataset.load_data_lookup
-
-    #    # TODO Perform checks on all from-to dimensions in needs_full_join.
+    @track_timing(timer_stats_collector)
+    def check_dataset_base_to_project_base_mappings(
+        self,
+        project_config: ProjectConfig,
+        dataset_config: DatasetConfig,
+        dataset: Dataset,
+        mapping_references,
+    ):
+        """Check that the dataset contains all mappings of base dimensions.
+
+        Parameters
+        ----------
+        project_config : ProjectConfig
+        dataset_config : DatasetConfig
+        dataset : Dataset
+
+        Raises
+        ------
+        DSGInvalidDimensionMapping
+            Raised if a requirement is violated.
+
+        """
+        # TODO: Handle this polymorphically when PR #99 is merged.
+        exclude_dims = set([DimensionType.TIME])
+        if dataset_config.model.data_schema_type == DataSchemaType.STANDARD:
+            exclude_dims.add(dataset_config.model.data_schema.load_data_column_dimension)
+        else:
+            raise Exception(f"not supported yet: {dataset_config.model.data_schema}")
+
+        needs_full_join = set()
+        types = [x for x in DimensionType if x not in exclude_dims]
+        for type1, type2 in itertools.product(types, types):
+            if type1 != type2:
+                needs_full_join.add((type1, type2))
+
+        for type1, type2 in needs_full_join:
+            if project_config.has_dimension_association(type1, type2):
+                records = project_config.get_dimension_association_records(type1, type2)
+            else:
+                pdim1 = project_config.get_base_dimension(type1)
+                pdim1_ids = pdim1.get_unique_ids()
+                pdim2 = project_config.get_base_dimension(type2)
+                pdim2_ids = pdim2.get_unique_ids()
+                records = itertools.product(pdim1_ids, pdim2_ids)
+                records = create_dataframe_from_dimension_ids(
```

Comment:
Tip: if `pdim1_ids` and `pdim12_ids` happen to be a col from other dataframes, you can do `pdim1_ids.crossJoin(pdim2_ids)` to get the product dataframe.